ciao-scheduler/scheduler.go

```diff
@@ -1000,6 +1003,12 @@ func configSchedulerServer() (sched *ssntpSchedulerServer) {
 	setLimits()
 
 	sched = newSsntpSchedulerServer()
+	payload, err := config.GetConfiguration(*configPath)
+	if err != nil {
+		glog.Errorf("Error loading configuration: %s", err)
+		return
+	}
+	sched.ssntp.SetClusterConfig(payload)
```

Comment:
The scheduler should not call into the configuration API, it should only pass the configuration information (URI and possibly a configuration driver - file, etcd, ...) to SSNTP.
To do the latter you should pass it as an ssntp.Config new field(s), so that when scheduler calls ssntp.Serve() then SSNTP gets that information and decides what to do with it, i.e. call into the configuration API.
By only keeping configuration package calls into the SSNTP implementation, you will not have to open code the above block for each and every SSNTP server one would implement.